src/os/bluestore/BlueStore.cc

```diff
@@ -10145,17 +10252,18 @@ int BlueStore::_upgrade_super()
   return 0;
 }
 
-void BlueStore::_assign_nid(TransContext *txc, OnodeRef o)
+bool BlueStore::_assign_nid(TransContext *txc, OnodeRef o)
```

Comment:
Instead of using the _assign_nid() return value to determine if a key is new, I suggest putting it directly in get_onode() in the create path, i.e. with
```
    // new object, new onode
    on = new Onode(this, oid, key);
```
that way we have just (1) the normal create path (there), (2) the remove case, and (3) the wonky clone and rename cases.

Reply:
done